refactor: separate search ui logic from frontend

Junjiequan opened the pull request:
## Description

This PR aims to separate the implementation of default user settings logic from FE.

## Motivation

Reduce maintenance costs for the future.

## Changes:


- Updated `frontend.config` to include new labelMaps and reorganize filter settings.
- Renamed and moved localColumns from `frontend.config`  into defaultDatasetsListSettings field.
- Removed unused interceptors for user settings.
- Added validation for filter components in findByIdUserSettings to check for valid filters based on FilterComponentType, and reset if invalid filters are found during validation.
- Adjusted findOneAndUpdateUserSettings to ensure proper upsert and default behavior

## Tests included

- [ ] Included for each change/fix?
- [ ] Passing? (Merge will not be approved unless this is checked) 

## Documentation
- [ ] swagger documentation updated \[required\]
- [ ] official documentation updated \[nice-to-have]

### official documentation info
If you have updated the official documentation, please provide PR # and URL of the pages where the updates are included

Junjiequan committed:
refacor: separate search ui logic from frontend

Junjiequan commented on src/users/schemas/user-settings.schema.ts:
```diff
@@ -74,12 +71,12 @@ export class UserSettings {
-    default: FILTER_CONFIGS,
+    default: [],
   })
   filters: FilterConfig[];
```
FilterConfig needs to be changed.
It should be similar to:
```suggestion
export type FilterComponentType =
  | "LocationFilterComponent"
  | "PidFilterComponent"
  | "PidFilterContainsComponent"
  | "PidFilterStartsWithComponent"
  | "GroupFilterComponent"
  | "TypeFilterComponent"
  | "KeywordFilterComponent"
  | "DateRangeFilterComponent"
  | "TextFilterComponent";

// Define the Filter interface
export interface FilterConfig {
  FilterComponentType: string;
  visible: boolean;
}
```

Junjiequan committed:
refactor: separate search UI logic from frontend

Junjiequan committed:
fix for failing test

Junjiequan committed:
fix default filter labelmaps

Junjiequan committed:
minor fixes

Junjiequan committed:
revert user settings interceptor and added reset user setting function

Junjiequan committed:
changed tap to swtichMap for asynchronous events for user-settings interceptor.
moved user-settings interceptor to login endpoint

nitrosx commented on src/users/users.controller.ts:
```diff
@@ -100,6 +94,7 @@ export class UsersController {
 
   @ApiBody({ type: CredentialsDto })
   @AllowAny()
+  @UseInterceptors(UserSettingsInterceptor)
```
I do not think we need this interceptor on login endpoint.

Junjiequan replied:
It's removed in the following commit

nitrosx commented on src/users/users.service.ts:
```diff
@@ -273,16 +274,27 @@ export class UsersService implements OnModuleInit {
   }
 
   async findByIdUserSettings(userId: string): Promise<UserSettings | null> {
```
_findUserSettingsById_ is a better name

nitrosx replied:
The renaming of the internal functions needs more brainstorming.
It requires more effort in order to keep naming consistent.

nitrosx commented on src/users/interceptors/user-settings.interceptor.ts:
Do we need this interceptor?

Junjiequan replied:
It's removed in the following commit.

Junjiequan committed:
refactor(auth): implement post-login tasks for user settings management and remove UserSettingsInterceptor

Junjiequan committed:
removed all user filter settings related logic

Junjiequan committed:
refactor(user-settings): consolidate user settings structure and remove deprecated fields

nitrosx commented on src/users/dto/update-user-settings.dto.ts:
```diff
@@ -18,13 +10,14 @@ export class UpdateUserSettingsDto {
+  })
+  @IsOptional()
+  @IsObject()
+  readonly frontendSettings?: Record<string, unknown[]>;
```
On second thought, can we call them just `extraSettings`?

Junjiequan replied:
ExtraSettings seems a bit too vague. I would kindly suggest reconsidering the name to something more specific.

nitrosx reviewed (changes requested):
Minor changes.

Junjiequan committed:
refactor(user-settings): update user settings structure and remove deprecated fields

This commit refactors the user settings structure by updating the field names and removing deprecated fields. The `frontendSettings` field has been renamed to `externalSettings` to better reflect its purpose. Additionally, the deprecated fields have been removed from the user settings schema.

Junjiequan committed:
fix unit test for users controller

Junjiequan committed:
changed login endpoint from userService to authService

Junjiequan committed:
added test for user settings patch
